snuba/query/snql/parser.py

```diff
@@ -1199,7 +1201,10 @@ def parse_snql_query(
         ],
     )
 
-    # Time based processing, which will sometimes be skipped by subscriptions
+    # Custom processing to tweak the AST before validation
+    _post_process(query, custom_processing)
```

Comment:
This might make more sense to move before the other post_processing, so the aliases are done automatically. Otherwise the condition below should alias the columns using the `_snuba_` prefix. The closer subscription queries are treated as regular queries the better it will be for us in the future.

If you do that, you can move the `_replace_time_condition` to the main list of `_post_process`, since there isn't a need for it to be separate now that there is a `custom_processing` section.

Reply:
I am not convinced about this. 
That would mean doing processing before _parse_subscriptables and before aliases are expanded or columns are qualified. Which is different than how we do processing everywhere else. I think we should not change the preconditions people need to take into account when writing processing logic. 

> Otherwise the condition below should alias the columns using the _snuba_ prefix.

I don't think this should ever happen. Alias mangling is meant only to avoid shadowing specifically on the SELECT clause. We would never add anything to the select clause (otherwise the output of the query would not match what the user expect) and every expression we add anywhere else is generally not aliased (as we do in query processors).

Reply:
OK that makes sense. Let's leave it as is.